here-naksha-storage-http/src/main/java/com/here/naksha/storage/http/HttpStorageReadExecute.java

```diff
@@ -41,49 +40,61 @@
 class HttpStorageReadExecute {
 
   private static final Logger log = LoggerFactory.getLogger(HttpStorageReadExecute.class);
+  private static final String HDR_STREAM_ID = "Stream-Id";
 
   @NotNull
-  static Result execute(ReadFeaturesProxyWrapper request, RequestSender sender) {
+  static Result execute(@NotNull NakshaContext context, ReadFeaturesProxyWrapper request, RequestSender sender) {
 
     return switch (request.getReadRequestType()) {
-      case GET_BY_ID -> executeFeatureById(request, sender);
-      case GET_BY_IDS -> executeFeaturesById(request, sender);
-      case GET_BY_BBOX -> executeFeatureByBBox(request, sender);
-      case GET_BY_TILE -> executeFeaturesByTile(request, sender);
+      case GET_BY_ID -> executeFeatureById(context, request, sender);
+      case GET_BY_IDS -> executeFeaturesById(context, request, sender);
+      case GET_BY_BBOX -> executeFeatureByBBox(context, request, sender);
+      case GET_BY_TILE -> executeFeaturesByTile(context, request, sender);
     };
   }
 
-  private static Result executeFeatureById(ReadFeaturesProxyWrapper readRequest, RequestSender requestSender) {
+  private static Result executeFeatureById(
+      @NotNull NakshaContext context, ReadFeaturesProxyWrapper readRequest, RequestSender requestSender) {
     String featureId = readRequest.getQueryParameter(FEATURE_ID);
 
-    HttpResponse<String> response =
-        requestSender.sendRequest(String.format("/%s/features/%s", baseEndpoint(readRequest), featureId));
+    HttpResponse<String> response = requestSender.sendRequest(
+        String.format("/%s/features/%s", baseEndpoint(readRequest), featureId),
+        Map.of(HDR_STREAM_ID, context.getStreamId()));
```

Comment:
I looks like HDR_STREAM_ID should be used every time in every case.
Also it looks the stream id in `NakshaContext` is final.

What about setting stream id in `RequestSender` then?
What I mean is:
1. Pass stream id in `RequestSender` constructor and keep it as a field
2. Add the stream id to the request in the `RequestSender.sendRequest` method

Pros:
- no code repetition
- no need to pass whole context up to executeFeatureBy... methods
- no need to remember about setting stream Id if  new executeFeatureBy... method needs to be created
- if there is a need to change something with stream id in the feature (e.g. null handling), there is one place that needs a change

Reply:
I see RequestSender as an eligible candidate for caching at HttpClient level (or in other words, at StorageId level). Passing any "per request" information to RequestSender at constructor level, will not allow us to effectively cache it the way I see it. agree?

Reply:
Agree